libs/domain/picking/src/mixins/picking-list.mixin.ts

```diff
@@ -0,0 +1,53 @@
+import { resolve, Type } from '@spryker-oryx/di';
+import {
+  asyncState,
+  isDefined,
+  observe,
+  valueType,
+} from '@spryker-oryx/utilities';
+import { LitElement } from 'lit';
+import { property } from 'lit/decorators.js';
+import {
+  BehaviorSubject,
+  distinctUntilChanged,
+  filter,
+  Observable,
+  switchMap,
+} from 'rxjs';
+import type { PickingList, PickingListComponentProperties } from '../models';
+import { PickingListService } from '../services';
+
+export declare class PickingListMixinInterface
+  implements PickingListComponentProperties
+{
+  protected pickingListService: PickingListService;
+  pickingListId?: string;
+  protected pickingList$: Observable<PickingList>;
+  protected pickingList: PickingList;
+}
+
+export const PickingListMixin = <
+  T extends Type<LitElement & PickingListComponentProperties>
+>(
+  superClass: T
+): Type<PickingListMixinInterface> & T => {
+  class AddressMixinClass extends superClass {
```

Comment:
Why it is called `AddressMixinClass`?

Reply:
Name is not important, but we need to provide some name for the class that is returned by mixin due to typescript limitation: https://lit.dev/docs/composition/mixins/#applying-decorators-in-mixins

Reply:
Let's name it more general like `MixinClass` because `AddressMixinClass` looks strange a bit